app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -89,9 +89,19 @@ def link_identity_to_service_provider
         current_user: current_user,
         ial: resolved_authn_context_int_ial,
         rails_session_id: session.id,
+        email_address_id: email_address_id,
       )
     end
 
+    def email_address_id
+      return session[:sp_email] unless session[:sp_email].nil?
```

Comment:
1. rename to `_id`
2. I think a positive check reads more clearly than `unless`
```suggestion
      return session[:sp_email_id] if session[:sp_email_id].present?
```

Reply:
Revised to clarify check